packages/server/graphql/public/types/Organization.ts

```diff
@@ -53,20 +52,27 @@ const Organization: OrganizationResolvers = {
     return getActiveTeamCountByOrgIds(orgId)
   },
 
-  allTeams: async ({id: orgId}, _args, {dataLoader, authToken}) => {
+  teams: async ({id: orgId}, _args, {dataLoader, authToken}) => {
     const viewerId = getUserId(authToken)
-    const [allTeamsOnOrg, isOrgAdmin] = await Promise.all([
+    const [teamsInOrg, isOrgAdmin] = await Promise.all([
       dataLoader.get('teamsByOrgIds').load(orgId),
-      dataLoader.get('organizations').loadNonNull(orgId),
       isUserOrgAdmin(viewerId, orgId, dataLoader)
     ])
-    const sortedTeamsOnOrg = allTeamsOnOrg.sort((a, b) => a.name.localeCompare(b.name))
+
     if (isOrgAdmin || isSuperUser(authToken)) {
-      const viewerTeams = sortedTeamsOnOrg.filter((team) => authToken.tms.includes(team.id))
-      const otherTeams = sortedTeamsOnOrg.filter((team) => !authToken.tms.includes(team.id))
-      return [...viewerTeams, ...otherTeams]
+      // Org admins and super users can see all teams
+      const viewerTeams = teamsInOrg.filter((team) => authToken.tms.includes(team.id))
+      const otherTeams = teamsInOrg.filter((team) => !authToken.tms.includes(team.id))
+
+      const sortedViewerTeams = viewerTeams.sort((a, b) => a.name.localeCompare(b.name))
+      const sortedOtherTeams = otherTeams.sort((a, b) => a.name.localeCompare(b.name))
+      return [...sortedViewerTeams, ...sortedOtherTeams]
     } else {
-      return sortedTeamsOnOrg.filter((team) => authToken.tms.includes(team.id))
+      // Regular users can see teams they're on plus public teams
+      const accessibleTeams = teamsInOrg.filter(
+        (team) => team.isPublic || authToken.tms.includes(team.id)
+      )
+      return accessibleTeams.sort((a, b) => a.name.localeCompare(b.name))
```

Comment:
-1 Why would we order the fields for admins but not for regular users?

Reply:
Initially, for org admins & super users, I was returning the viewer teams and then the teams they're not a member of, but I moved that logic to the client. That code was left behind. I've now removed it.